### PR TITLE
[fix](jdbc catalog) fix JdbcScanNode `NOT` CompoundPredicate filter expr handling errors

### PR DESCRIPTION
**Pull request by zy-kkk**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zy-kkk** [comment]
run buildall

**Review by morningman — DISMISSED** [review]
LGTM

File: fe/fe-core/src/main/java/org/apache/doris/planner/external/jdbc/JdbcScanNode.java [hunk 1]
```diff
@@ -328,13 +328,31 @@ public static String conjunctExprToString(TOdbcTableType tableType, Expr expr, J
         if (expr instanceof CompoundPredicate) {
             StringBuilder result = new StringBuilder();
             CompoundPredicate compoundPredicate = (CompoundPredicate) expr;
+
+            // If the operator is 'NOT', prepend 'NOT' to the start of the string
+            if ("NOT".equals(compoundPredicate.getOp().toString())) {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/planner/external/jdbc/JdbcScanNode.java:333]: i think u should use `compoundPredicate.op == Operator.NOT`

**Comment by zy-kkk** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**